pkg/lso/localvolumediscovery.go

```diff
@@ -0,0 +1,227 @@
+func PullLocalVolumeDiscovery(apiClient *clients.Settings, name, nsname string) (*LocalVolumeDiscoveryBuilder, error) {
+	glog.V(100).Infof(
+		"Pulling localVolumeDiscovery object name:%s in namespace: %s", name, nsname)
+
+	builder := LocalVolumeDiscoveryBuilder{
+		apiClient: apiClient,
+		Definition: &lsoV1alpha1.LocalVolumeDiscovery{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the localVolumeDiscovery is empty")
+
+		builder.errorMsg = "localVolumeDiscovery 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the localVolumeDiscovery is empty")
+
+		builder.errorMsg = "localVolumeDiscovery 'nsname' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("localVolumeDiscovery oject %s doesn't exist in namespace %s", name, nsname)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Get fetches existing localVolumeDiscovery from cluster.
+func (builder *LocalVolumeDiscoveryBuilder) Get() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Pulling existing localVolumeDiscovery with name %s under namespace %s from cluster",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	lvd := &lsoV1alpha1.LocalVolumeDiscovery{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name:      builder.Definition.Name,
+		Namespace: builder.Definition.Namespace,
+	}, lvd)
+
+	builder.Object = lvd
+
+	return err
+}
+
+// Exists checks whether the given localVolumeDiscovery exists.
+func (builder *LocalVolumeDiscoveryBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if localVolumeDiscovery %s exists in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	err := builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// IsDiscovering check if the localVolumeDiscovery is Discovering.
+func (builder *LocalVolumeDiscoveryBuilder) IsDiscovering() (bool, error) {
+	if valid, err := builder.validate(); !valid {
+		return false, err
+	}
+
+	glog.V(100).Infof("Verify localVolumeDiscovery %s in namespace %s is successful",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if !builder.Exists() {
+		return false, fmt.Errorf("localVolumeDiscovery %s not found in %s namespace",
+			builder.Definition.Name, builder.Definition.Namespace)
+	}
+
+	phase, err := builder.GetPhase()
+
+	if err != nil {
+		return false, fmt.Errorf("failed to get phase value for %s localVolumeDiscovery in %s namespace due to %w",
+			builder.Definition.Name, builder.Definition.Namespace, err)
+	}
+
+	if phase == "Discovering" {
+		return true, nil
+	}
+
+	return false, fmt.Errorf("bad %s localVolumeDiscovery in %s namespace phase: %s",
```

Comment:
Please update log to invalid %s localVolumeDiscovery phase in %s namespace phase: %s"